src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -100,6 +124,19 @@ export class TechRecordSummaryComponent implements OnInit {
     this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
   }
 
+  findAxleToRemove(axles: Axle[]): number {
+    let previousAxleRow = 1;
+
+    for (const ax of axles) {
+      if (ax.axleNumber === previousAxleRow) {
+        previousAxleRow += 1;
+      } else {
+        return ax.axleNumber! - 1;
+      }
+    }
+    return axles.length + 1;
+  }
+
```

Comment:
```suggestion
  findAxleToRemove(axles: Axle[]): number {
    const incorrectAxle = axles.find(axle => axle.axleNumber !== axles.indexOf(axle) + 1)
    return incorrectAxle ? incorrectAxle.axleNumber! - 1 : axles.length + 1;
  }

```
Don't really know what this function is intended for but this is probably a cleaner way to implement it.

Reply:
It's part of Nathan and Tom's axle ticket, they wanted to pick out a specific axle and remove one rather than pop/unshift from an array. I don't know much more than that.

Reply:
I believe the remove event only emits the remaining axles. This function works out which was removed.